hadoop-crypto/src/test/java/com/palantir/hadoop/ConfigurableEncryptedFileSystemTest.java

```diff
@@ -0,0 +1,141 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.hadoop;
+
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.not;
+import static org.junit.Assert.assertThat;
+import static org.junit.Assert.assertTrue;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.net.URI;
+import java.nio.charset.StandardCharsets;
+import java.security.KeyPair;
+import org.apache.commons.io.IOUtils;
+import org.apache.commons.net.util.Base64;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.fs.RawLocalFileSystem;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.junit.rules.TemporaryFolder;
+
+public final class ConfigurableEncryptedFileSystemTest {
+
+    private static final URI EFS_URI = URI.create("efs:///");
+
+    private FileSystem efsPublic;
+    private FileSystem efs;
+    private FileSystem rawFs;
+    private Path path;
+
+    @Rule
+    public TemporaryFolder folder = new TemporaryFolder();
+
+    @Rule
+    public ExpectedException expectedException = ExpectedException.none();
+
+    @Before
+    public void before() throws IOException {
+        KeyPair keyPair = TestKeyPairs.generateKeyPair();
+        Configuration conf = getBaseConf();
+        conf.set(ConfigurableEncryptedFileSystem.PUBLIC_KEY_CONF,
+                Base64.encodeBase64String(keyPair.getPublic().getEncoded()));
+        conf.set(ConfigurableEncryptedFileSystem.BACKING_FILESYSTEM_CONF,
+                RawLocalFileSystem.class.getCanonicalName());
+
+        efsPublic = FileSystem.newInstance(EFS_URI, conf);
+
+        conf.set(ConfigurableEncryptedFileSystem.PRIVATE_KEY_CONF,
+                Base64.encodeBase64String(keyPair.getPrivate().getEncoded()));
+
+        efs = FileSystem.newInstance(EFS_URI, conf);
+        rawFs = FileSystem.newInstance(URI.create("file:///"), conf);
+        path = new Path(folder.newFile().getAbsolutePath());
+    }
+
+    @Test
+    public void testReadWrite() throws IOException {
+        String data = "data";
+        byte[] dataBytes = data.getBytes(StandardCharsets.UTF_8);
+        byte[] readData = new byte[data.length()];
+
+        // Write encrypted data
+        OutputStream os = efs.create(path);
+        IOUtils.write(data, os);
+        os.close();
+
+        // Read encrypted data
+        InputStream dis = efs.open(path);
+        IOUtils.readFully(dis, readData);
+        assertThat(readData, is(dataBytes));
```

Comment:
yeah. `equalTo` check to see if the type is an array and then does an array comparison. the test below this correctly checks that the arrays are NOT equal so that gives me confidence its doing the right thing